matchms/filtering/derive_formula_from_name.py

```diff
@@ -0,0 +1,51 @@
+import re
+from ..typing import SpectrumType
+
+
+def derive_formula_from_name(spectrum_in: SpectrumType, remove_formula_from_name=True) -> SpectrumType:
+    """Detect and remove misplaced formula in compound name and add to metadata.
+
+    Method to find misplaced formulas in compound name based on regular expression.
+    Args:
+    ----
+    spectrum_in: SpectrumType
+        Input spectrum.
+    remove_formula_from_name: bool
+        Remove found formula from compound name if set to True. Default is True.
+    """
+    if spectrum_in is None:
+        return None
+
+    spectrum = spectrum_in.clone()
+
+    # Get compound name
+    if spectrum.get("compound_name", None) is not None:
+        name = spectrum.get("compound_name")
+    else:
+        assert spectrum.get("name", None) in [None, ""], ("Found 'name' but not 'compound_name' in metadata",
+                                                          "Apply 'add_compound_name' filter first.")
+        return spectrum
+
+    # Detect formula at end of compound name
```

Comment:
Consider the shorter:
end_of_name = name.split(" ")[-1]
formula_from_name = end_of_name if looks_like_formula(end_of_name) else None

Reply:
Yes, thanks. I changed it to your suggestion.